### PR TITLE
Add Dependency Injection support to Real Service Pipeline for DI of Msft-internal packages

### PR DESCRIPTION
**Pull request by markfields**
This adds a "fake" package under the `tools/pipelines` dir called `dependency-injection`, which includes dependencies on any internal packages we want available for DI during pipeline runs.  A pipeline can easily install that package and then code can dynamically load the installed dependency that it wants.

For the first use case, the package injected will contain the Aria logger that emulates Office's Fluid Logger.  See https://dev.azure.com/fluidframework/internal/_git/ff_internal/pullrequest/15?_a=overview for the latest version of the package - just a stub that imports the required dependencies and adds a test field to `global`.

Next step after this PR is to update that package to implement a function called like `getTestLogger`, and then call that function before each `end-to-end` test and inject that logger throughout.

### CODE REVIEW

File: packages/test/service-load-test/scripts/_package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "dependency-injection",
+  "version": "0.0.1",
+  "description": "Inject dependencies into pipeline VM",
+  "main": "index.js",
```
Comment (markfields) [inline packages/test/service-load-test/scripts/_package.json:5]: I wonder if I can remove this

**Comment by markfields** [comment]
Yay it's working - I was able to inject a thing from @oteljs into our mocha tests!  See [these](https://dev.azure.com/fluidframework/internal/_build/results?buildId=16246&view=ms.vss-test-web.build-test-results-tab&runId=42068&resultId=100001&paneView=debug )results.  Going to clean up the PR and get it merged!

File: tools/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ const urlPattern = /(https?[^"@]+)(\/@.+|\/[^/]+\/-\/.+tgz)/g;
 
 export const handler: Handler = {
     name: "package-lockfiles",
-    match: /^.*?package-lock\.json$/i,
+    match: /^.*?[^_]package-lock\.json$/i, // Ignore _package-lock.json
```
Comment (markfields) [inline tools/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts:16]: Covered by https://github.com/microsoft/FluidFramework/pull/5170, will need to merge that then update build tools version in this PR...

File: packages/test/end-to-end-tests/src/test/.mocharc.js [hunk 3]
```diff
@@ -9,14 +9,21 @@ const testDriver = process.env.FLUID_TEST_DRIVER ? process.env.FLUID_TEST_DRIVER
 const packageDir = `${__dirname}/../..`;
 const testPackagesDir = `${packageDir}/..`;
 
-const config ={
+const requiredModules = [
+    `${testPackagesDir}/mocha-test-setup`, // General mocha setup e.g. suppresses console.log
+    `${testPackagesDir}/test-drivers`, // Inject implementation of getFluidTestDriver, configured via FLUID_TEST_DRIVER
+];
+
+if (process.env.FLUID_DI_ROOT && process.env.FLUID_DI_LOGGERPKG) {
```
Comment (anthony-murphy) [inline packages/test/end-to-end-tests/src/test/.mocharc.js:17]: we should consider just making this a single var, like FUILD_TEST_LOGGER_PKG_PATH

**Review by anthony-murphy — APPROVED** [review]

File: packages/test/end-to-end-tests/src/test/.mocharc.js (hunk 3 above)
Comment (markfields) [reply packages/test/end-to-end-tests/src/test/.mocharc.js:17]: Haha yeah I have it that way in my next branch :P

**Pull request merged**